assignments/array-methods.js

```diff
@@ -56,28 +56,77 @@ const runners = [{"id":1,"first_name":"Charmain","last_name":"Seiler","email":"c
 // ==== Challenge 1: Use .forEach() ====
 // The event director needs both the first and last names of each runner for their running bibs.  Combine both the first and last names into a new array called fullName. 
 let fullName = [];
+    runners.forEach(function(runners) {
+        fullName.push(runners.first_name + ' ' + runners.last_name);
+    });
+
 console.log(fullName);
 
 // ==== Challenge 2: Use .map() ====
 // The event director needs to have all the runner's first names converted to uppercase because the director BECAME DRUNK WITH POWER. Convert each first name into all caps and log the result
-let allCaps = [];
+let allCaps = runners.map((person) => {
+    return person.first_name.toUpperCase();
+    });
 console.log(allCaps); 
 
 // ==== Challenge 3: Use .filter() ====
 // The large shirts won't be available for the event due to an ordering issue.  Get a list of runners with large sized shirts so they can choose a different size. Return an array named largeShirts that contains information about the runners that have a shirt size of L and log the result
-let largeShirts = [];
+let largeShirts = runners.filter((runner) => {
+    return runner.shirt_size === 'L';
+});
+
 console.log(largeShirts);
 
 // ==== Challenge 4: Use .reduce() ====
 // The donations need to be tallied up and reported for tax purposes. Add up all the donations into a ticketPriceTotal array and log the result
-let ticketPriceTotal = [];
+let ticketPriceTotal = runners.reduce(function(donation, runners) {
+    return donation += runners.donation;
+});
+
+let ticketPTotal = runners.reduce((donation, runners) => {
+    return total =+ runners.donation;
+}, 0);
+
+console.log(ticketPTotal);
+// this is wrong
+
 console.log(ticketPriceTotal);
+// this is wrong
 
 // ==== Challenge 5: Be Creative ====
 // Now that you have used .forEach(), .map(), .filter(), and .reduce().  I want you to think of potential problems you could solve given the data set and the 5k fun run theme.  Try to solve 3 unique problems using one or many of the array methods listed above.
 
 // Problem 1
+// Every runner who donated $200 or more gets a special prize. A list of everyone who donated $200 or greater needs to be made.
+
+let largeDonors = runners.filter((runner) => {
+    if (runner.donation >= 200) {
+        return runner.first_name;
+    }
+});
+
+console.log(largeDonors);
 
 // Problem 2
+// A list of all emails and names needs to be made so all participants may be emaild. Make a list of first names and email.
+
+let emailListWithNames = [];
+runners.forEach(function(runners) {
+    emailListWithNames.push(runners.first_name + ", " + runners.email);
+});
+
+console.log(emailListWithNames);
+
+// Problem 3
+// Wordtune is sponsering each of it's racers and is matching donations. Make a list of all donors who work at Wordtune as well as the amount of their donation.
+
+let matchedDonations = [];
+
+runners.filter((runners) => {
```

Comment:
I'm not sure filter is the best use for what this is trying to accomplish, though I like how you approached the problem. Filter is good for excluding elements from an array while the remaining elements stay the same. I would use `.filter` to get an array of only Wordtune employees, then use `.map` or `.forEach` to get the array of Wordtune employees + their donations.

Here's an example. Let's say I have a list of numbers, 1-10. I want a list of all the even numbers multiplied by 2.
```
const numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

const onlyEvenNumbers = numbers.filter(function(number) {
  return number % 2 === 0            // this checks if number is evenly divisible by 2
})

console.log(onlyEvenNumbers);    // this logs out [2, 4, 6, 8, 10]

const doubledEvenNumbers = onlyEvenNumbers.map(function(number) {
  return number * 2;
})

console.log(doubledEvenNumbers);    // this logs out [4, 8, 12, 16, 20]
```

See how you could approach this problem in the same way.